### PR TITLE
Split input_data into three modules for atm, sfc, and nst.

### LINKED ISSUE #460: Split input_data.F90 into `atm_input_data`, `sfc_input_data`, and `nst_target_data`
`input_data.F90` is a really long file that I think could be broken up to make it more managable. It seems like it does 3 different things for 3 types of data, and it would make sense to group them together.

### PR DESCRIPTION
**Pull request by kgerheiser**
Shared routines and data are stored in input_data_utils.

Fixes #460

### CODE REVIEW

File: sorc/chgres_cube.fd/input_data_utils.F90 [hunk 1]
```diff
@@ -0,0 +1,178 @@
+!> @file
+!! @brief Shared routines and data for atmosphere, surface, and nst input_data modules.
+!! @author George Gayno NCEP/EMC
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/input_data_utils.F90:3]: Add your name as a file author for all these .F90 files.

**Comment by kgerheiser** [comment]
This and #482 conflict. Once that is merged I will bring the updates into this PR.

File: sorc/chgres_cube.fd/atm_input_data.F90 [hunk 2]
```diff
@@ -0,0 +1,2844 @@
...
+!! associated with the input grid.
+!!
+!! @author George Gayno, Kyle Gerheiser NCEP/EMC
+module atm_input_data_mod
+  use esmf
+  use netcdf
+  use nemsio_module
+
+
+  use program_setup, only          : data_dir_input_grid, &
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/atm_input_data.F90:20]: Check the 'use' statements in the new modules. Most of these variables are not used.

Comment (kgerheiser) [reply sorc/chgres_cube.fd/atm_input_data.F90:20]: Ah, I meant to do that

**Comment by edwardhartnett** [comment]
Is this ready to be merged?

**Comment by kgerheiser** [comment]
I think #482 should go in first and then I'll fix the conflict that will arise

**Comment by LarissaReames-NOAA** [comment]
> I think #482 should go in first and then I'll fix the conflict that will arise

Any updates on getting these conflicts resolved? It might be good to go ahead and get this change in since I imagine there will be several PR's modifying this file in the coming weeks.

**Comment by kgerheiser** [comment]
I will just redo this PR and open a new one

**Pull request closed without merging**